EssentialCSharp.Web.Tests/SitemapXmlHelpersTests.cs

```diff
@@ -0,0 +1,285 @@
+                tempDir,
+                siteMappings,
+                routeConfigurationService,
+                baseUrl,
+                out var nodes);
+
+            var allUrls = nodes.Select(n => n.Url).ToList();
+
+            // Should not include Index action routes (they're the default)
+            Assert.DoesNotContain(allUrls, url => url.Contains("/Index", StringComparison.OrdinalIgnoreCase));
+        });
+    }
+
+    [Fact]
+    public void GenerateSitemapXml_DoesNotIncludeErrorRoutes()
+    {
+        // Arrange
+        var tempDir = new DirectoryInfo(Path.GetTempPath());
+        var siteMappings = new List<SiteMapping>();
+        var baseUrl = "https://test.example.com/";
+
+        // Act & Assert
+        _Factory.InServiceScope(serviceProvider =>
+        {
+            var routeConfigurationService = serviceProvider.GetRequiredService<IRouteConfigurationService>();
+            SitemapXmlHelpers.GenerateSitemapXml(
+                tempDir,
+                siteMappings,
+                routeConfigurationService,
+                baseUrl,
+                out var nodes);
+
+            var allUrls = nodes.Select(n => n.Url).ToList();
+
+            // Should not include Error action routes
+            Assert.DoesNotContain(allUrls, url => url.Contains("/Error", StringComparison.OrdinalIgnoreCase));
+        });
+    }
+
+    [Fact]
+    public void GenerateAndSerializeSitemapXml_CreatesFileSuccessfully()
+    {
+        // Arrange
+        var logger = _Factory.Services.GetRequiredService<ILogger<SitemapXmlHelpersTests>>();
+        var tempDir = new DirectoryInfo(Path.GetTempPath());
+        var siteMappings = new List<SiteMapping> { CreateSiteMapping(1, 1, true) };
+        var baseUrl = "https://test.example.com/";
+
+        // Clean up any existing file
+        var expectedXmlPath = Path.Combine(tempDir.FullName, "sitemap.xml");
+        if (File.Exists(expectedXmlPath))
+        {
+            File.Delete(expectedXmlPath);
+        }
+
+        try
+        {
+            // Act
+            _Factory.InServiceScope(serviceProvider =>
+            {
+                var routeConfigurationService = serviceProvider.GetRequiredService<IRouteConfigurationService>();
+                SitemapXmlHelpers.GenerateAndSerializeSitemapXml(
+                    tempDir,
+                    siteMappings,
+                    logger,
+                    routeConfigurationService,
+                    baseUrl);
+            });
+
+            // Assert
+            Assert.True(File.Exists(expectedXmlPath));
+
+            var xmlContent = File.ReadAllText(expectedXmlPath);
+            Assert.Contains("<?xml", xmlContent);
+            Assert.Contains("<urlset", xmlContent);
+            Assert.Contains(baseUrl, xmlContent);
+        }
+        finally
+        {
+            // Clean up
+            if (File.Exists(expectedXmlPath))
+            {
+                File.Delete(expectedXmlPath);
+            }
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.io.file.delete?view=net-8.0#remarks
```suggestion
            File.Delete(expectedXmlPath);
```